docs/docsite/rst/shared_snippets/variable_evaluation.rst

```diff
@@ -0,0 +1,30 @@
+.. _variable_evaluation:
+
+Variable Evaluation
+```````````````````
+
+In general, Ansible evaluates any variables in playbook content at the last
+possible second, which means that if you define a data structure that data
+structure itself can define variable values within it, and everything “just
+works” as you would expect. This also means variable strings can include other
```

Comment:
I disagree with the phrase `"just works" as you would expect`: it is subjective and does not illuminate the actual behaviour.  In my case, the fact that lazy evaluation is used came as a complete surprise.